react-validated-input-component/my-app/src/ValidatedInput.js

```diff
@@ -0,0 +1,78 @@
+import { useState } from "react"
+import './ValidatedInput.css'
+
+export default function ValidatedInput() {
+  const [password, setPassword] = useState('');
+
+  // let errorMessage;
+  // const passwordValidation = /.{1,}/;
+  // const passwordValidationResult = passwordValidation.test(password);
+  // const lengthValidation = /.{8,}/;
+  // const lengthResult = lengthValidation.test(password);
+  // const numberValidation = /\d/;
+  // const numberValidationResult = numberValidation.test(password);
+  // const uppercaseValidation = /[A-Z]/;
+  // const uppercaseValidationResult = uppercaseValidation.test(password);
+  // const specialValidation = /[!@#$%^&*()]/;
+  // const specialValidationResult = specialValidation.test(password);
+
+  // switch (false) {
+  //   case passwordValidationResult: errorMessage = 'A password is required.'; break;
+  //   case lengthResult: errorMessage = 'Your password is too short.'; break;
+  //   case numberValidationResult: errorMessage = 'Your password must contain a digit.'; break;
+  //   case uppercaseValidationResult: errorMessage = 'Your password must contain a capital letter.'; break;
+  //   case specialValidationResult: errorMessage = 'Your password must contain a special character ( ! ,  @ ,  # ,  $ ,  % ,  ^ ,  & ,  * ,  ( , or  ) ).'; break;
+  //   default: errorMessage = ''; break;
+  // }
+
+  let errorMessage = '';
+  const validationData = [
```

Comment:
This is nifty and easily extensible to other rules. In fact, you could now pass the validation rules as a prop, making this component even more reusable.